### PR TITLE
Add additional checks for read-only cell

### PR DESCRIPTION
**Pull request by pnowak**
### Context
Shouldn't throw an error `Cannot read property 'removeEventListener' of undefined` when trying `setDataAtCell` after selection on cell read-only or header.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6214
2. #6246
3. #6209
4. #5126

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by pnowak** [comment]
@jansiegel I chose the second one options - if the `AutoResize` plugin isn't actually observing anything, I don't calling the `unObserve` method.

**Comment by pnowak** [comment]
I'm not sure. We check `cellProperty.readOnly` in `prepare` method before we use `refreshDimensions`. So checking `cellProperty` in this method, in my opinion, isn't the good flow. 
The `observe` isn't private property, is exposed in the returned object. Isn't it?

And yes, the bug is still present.

File: lib/autoResize/autoResize.js [hunk 1]
```diff
@@ -173,7 +173,7 @@ function autoResize() {
       unObserve(el, 'compositionend', delayedResize);
     },
     resize: resize
-  };
+  }
```
Comment (swistach) [inline lib/autoResize/autoResize.js]:
```suggestion
  };
```

File: src/editorManager.js [hunk 2]
```diff
@@ -429,7 +432,13 @@ class EditorManager {
...
             this.activeEditor.enableFullEditMode();
           }
-          this.openEditor(null, event);
+
+          if (this.cellProperties.readOnly) {
+            this.moveSelectionAfterEnter();
+
+          } else {
+            this.openEditor(null, event);
+          }
```
[the remark this answers is not in the document]
Comment (pnowak) [reply src/editorManager.js]: Makes sense, thanks. I change this snippet.

File: src/editorManager.js [hunk 3]
```diff
@@ -215,6 +221,15 @@ class EditorManager {
     return this.closeEditor(true, isCtrlPressed);
   }
 
+  /**
+   * Deactivate active editor.
```
Comment (swistach) [inline src/editorManager.js]:
```suggestion
   * Clears reference to an instance of the active editor.
```

**Pull request merged**